apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/state.ex

```diff
@@ -17,35 +17,72 @@
 #
 
 defmodule Astarte.DataUpdaterPlant.DataUpdater.State do
-  defstruct [
-    :realm,
-    :device_id,
-    :message_tracker,
-    :introspection,
-    :groups,
-    :interfaces,
-    :interface_ids_to_name,
-    :interfaces_by_expiry,
-    :mappings,
-    :paths_cache,
-    :device_triggers,
-    :data_triggers,
-    :volatile_triggers,
-    :introspection_triggers,
-    :connected,
-    :total_received_msgs,
-    :total_received_bytes,
-    :initial_interface_exchanged_bytes,
-    :initial_interface_exchanged_msgs,
-    :interface_exchanged_bytes,
-    :interface_exchanged_msgs,
-    :last_seen_message,
-    :last_device_triggers_refresh,
-    :last_groups_refresh,
-    :datastream_maximum_storage_retention,
-    :trigger_id_to_policy_name,
-    :discard_messages,
-    :last_deletion_in_progress_refresh,
-    :last_datastream_maximum_retention_refresh
-  ]
+  use TypedStruct
+
+  @type interfaces_map() :: %{String.t() => Astarte.Core.Interface.t()}
```

Comment:
to add on what Omar said, "String" doesn't give mouch context on its own
Something like

```suggestion
  @type interface_name :: String.t()
  @type descriptors_by_name :: %{interface_name => InterfaceDescriptor.t()}
```

gives more insight about the meaning of this field